Service/src/main/java/ticket/Ticket.java

```diff
@@ -0,0 +1,253 @@
+        this.concertHall = "UNDEFINED";
+        this.day = LocalDate.of(2000, 01,01);
+        this.time = LocalTime.of(00,00,00);
+        this.isPromo = ticket.PromotionAvaliabilities.NOINFO;
+        this.stadiumSector = '\u0000';
+        this.maxWeight = 0;
+        this.ticketType = ticketTypes.NOINFO;
+    };
+
+    public Ticket(String concertHall, short eventCode, LocalDate day, LocalTime time){
+        this.ID = generateId();
+        if(eventCode<100 || eventCode>999){
+            throw new IllegalArgumentException("Enter 3-digit code");
+        }
+        this.eventCode = eventCode;
+        this.creationDate = LocalDate.now();
+        this.creationTime = LocalTime.now();
+        if(concertHall.length()>10){
+            throw new IllegalArgumentException("The name of the concert hall should not exceed 10 characters");
+        }
+        this.concertHall = concertHall;
+        this.day = day;
+        this.time = time;
+        this.isPromo = ticket.PromotionAvaliabilities.NOINFO;
+        this.stadiumSector = '\u0000';
+        this.maxWeight = 0;
+        this.ticketType = ticketTypes.NOINFO;
+    }
+
+    public Ticket(LocalDate creationDate, LocalTime creationTime, ticketTypes ticketType, BigDecimal cost){
+        this.ID = generateId();
+        this.eventCode = 0;
+        this.creationDate = creationDate;
+        this.creationTime = creationTime;
+
+        this.day = LocalDate.now();
+        this.time = LocalTime.now();
+        this.isPromo = ticket.PromotionAvaliabilities.NOINFO;
+        this.stadiumSector = '\u0000';
+        this.maxWeight = 0;
+        this.cost = cost;
+        this.ticketType = ticketType;
+    }
+
+    public Ticket(short eventCode, String concertHall, LocalDate day, LocalTime time,
+                  boolean isPromo, char stadiumSector, double maxWeight, BigDecimal cost, int ticketType){
+        this.ID = generateId();
+        if(eventCode<100 || eventCode>999) {
+            throw new IllegalArgumentException("Enter 3-digit code");
+        }
+        this.eventCode = eventCode;
+        this.creationDate = LocalDate.now();
+        this.creationTime = LocalTime.now();
+        if(concertHall.length()>10){
+            throw new IllegalArgumentException("The name of the concert hall should not exceed 10 characters");
+        }
+        this.concertHall = concertHall;
+        this.day = day;
+        this.time = time;
+        if(isPromo){
+            this.isPromo = ticket.PromotionAvaliabilities.YES;
+
+        }else{
+            this.isPromo = ticket.PromotionAvaliabilities.NO;
+
+        }
+        if(stadiumSector != 'A' && stadiumSector != 'B' && stadiumSector != 'C'){
+            throw new IllegalArgumentException("The name of sectors must be A, B or C");
+        }
+        this.stadiumSector = stadiumSector;
+        this.maxWeight = maxWeight;
+        this.cost = cost;
+        if(ticketType == 0){
+            this.ticketType = ticketTypes.NOINFO;
```

Comment:
if we have correspondence between type and number - we could add int field into Enum